File: agent/models/icp.py
from pydantic import BaseModel, Field


class ICPProfile(BaseModel):
    industries: list[str]
    company_size_range: dict = Field(alias="companySizeRange", default_factory=dict)
    revenue_range: dict = Field(alias="revenueRange", default_factory=dict)
    geographies: list[str] = []
    tech_stack: list[str] = Field(alias="techStack", default_factory=list)
    keywords: list[str] = []
    additional_notes: str = Field(alias="additionalNotes", default="")

    model_config = {"populate_by_name": True}


class BANTWeights(BaseModel):
    budget: float = Field(ge=0, le=1)
    authority: float = Field(ge=0, le=1)
    need: float = Field(ge=0, le=1)
    timeline: float = Field(ge=0, le=1)

File: frontend/src/types/icp.ts
export interface RangeValue {
  min: number;
  max: number;
}

export interface ICPProfile {
  industries: string[];
  companySizeRange: RangeValue;
  revenueRange: RangeValue;
  geographies: string[];
  techStack: string[];
  keywords: string[];
  additionalNotes: string;
}

export interface BANTWeights {
  budget: number;
  authority: number;
  need: number;
  timeline: number;
}

export interface ICPFormData {
  icpProfile: ICPProfile;
  bantWeights: BANTWeights;
  maxResults: number;
}
